src/common/matmul_pd.hpp

```diff
@@ -195,7 +207,13 @@ struct matmul_pd_t : public primitive_desc_t {
                                 sc.group_dims_[0] == 1
                                         && K() % sc.group_dims_[1] == 0);
             } else {
-                ok = ok && (mask == 0);
+                ok = ok
```

Comment:
Here must be a check for fp8 versus classic quantization.

Reply:
I dont think it is required. With these changes gemm supports dst scales whenever dst is an integer or fp8 so there should be no difference in treatment. Attempts to use dst scales with non-fp8 float dst will be filtered in attr checks.